### PR TITLE
feat: Added initial observability service implementation

### PR DESCRIPTION
**Pull request by JigarJoshi**

### CODE REVIEW

**Review by efirs — APPROVED** [review]

File: server/services/v1/observability.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+	Series   [1]struct {
+		PointList [][2]float64 `json:"pointlist"`
+	}
+}
+
+func (dd Datadog) QueryTimeSeriesMetrics(ctx context.Context, req *api.QueryTimeSeriesMetricsRequest) (*api.QueryTimeSeriesMetricsResponse, error) {
+	if !isAllowedMetric(req.MetricName, &config.DefaultConfig) {
+		return nil, api.Errorf(api.Code_PERMISSION_DENIED, "Failed to query metrics: reason = not allowed to query metric: "+req.MetricName)
+	}
+	ddReq, err := http.NewRequest(http.MethodGet, dd.ObservabilityConfig.ProviderUrl+"/api/v1/query", nil)
```
Comment (rbarabas) [inline server/services/v1/observability.go:70]: nit: should we reference /api/v1/query through a constant?

**Review by rbarabas — APPROVED** [review]
LGTM

File: server/services/v1/observability.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+type MetricsQueryResp struct {
+	FromDate int64 `json:"from_date"`
+	ToDate   int64 `json:"to_date"`
+	Query    string
+	Series   [1]struct {
+		PointList [][2]float64 `json:"pointlist"`
+	}
+}
+
+func (dd Datadog) QueryTimeSeriesMetrics(ctx context.Context, req *api.QueryTimeSeriesMetricsRequest) (*api.QueryTimeSeriesMetricsResponse, error) {
```
Comment (himank) [inline server/services/v1/observability.go:66]: How easy it is to use this in the Quota?  The use-case is to fetch a few metrics in the quota to make a decision about the throttling. If we can just utilize this code that would be ideal. @JigarJoshi @efirs  (that may not be timeseries though)

**Review by himank — APPROVED** [review]

**Comment by JigarJoshi** [comment]
Closing this PR and moving this into https://github.com/tigrisdata/tigris/pull/430 with newer changes (flexible query formation).

File: server/services/v1/observability.go (hunk 1 above)
Comment (JigarJoshi) [reply server/services/v1/observability.go:70]: took care of that here https://github.com/tigrisdata/tigris/pull/430

**Pull request closed without merging**